packages/market-service/src/coincap/coincap.ts

```diff
@@ -0,0 +1,151 @@
+import { adapters } from '@shapeshiftoss/caip'
+import { fromCAIP19 } from '@shapeshiftoss/caip/dist/caip19/caip19'
+import {
+  FindAllMarketArgs,
+  HistoryData,
+  HistoryTimeframe,
+  MarketCapResult,
+  MarketData,
+  MarketDataArgs,
+  PriceHistoryArgs
+} from '@shapeshiftoss/types'
+import axios from 'axios'
+import omit from 'lodash/omit'
+import dayjs from 'dayjs'
+
+import { MarketService } from '../api'
+import { CoinCapMarketCap } from './coincap-types'
+
+export class CoinCapMarketService implements MarketService {
+  baseUrl = 'https://api.coincap.io/v2'
+
+  private readonly defaultGetByMarketCapArgs: FindAllMarketArgs = {
+    pages: 10,
+    perPage: 250
+  }
+
+  findAll = async (args?: FindAllMarketArgs) => {
+    const argsToUse = { ...this.defaultGetByMarketCapArgs, ...args }
+    const { pages, perPage } = argsToUse
+    const urlAtPage = (page: number) =>
+    `${this.baseUrl}/assets?limit=${perPage}&offset=${page}`
+    const pageCount = Array(pages)
+      .fill(0)
+      .map((_v, i) => i + 1)
+
+    try {
+      const combined = (
+        await Promise.all(
+          pageCount.map(async (page) => axios.get<{ data: CoinCapMarketCap[] }>(urlAtPage(page)))
+        )
+      ).flat()
+
+      return combined
+        .map(({ data }) => data && data.data ? data.data : []) // filter out rate limited results
+        .flat()
+        .sort((a, b) => a.rank > b.rank ? 1 : -1)
+        .reduce((acc, cur) => {
+          const { id } = cur
+          try {
+            const caip19 = adapters.coingeckoToCAIP19(id)
+            const curWithoutId = omit(cur, 'id') // don't leak this through to clients
+            acc[caip19] = {
+              price: curWithoutId.priceUsd.toString(),
+              marketCap: curWithoutId.marketCapUsd.toString(),
+              volume: curWithoutId.volumeUsd24Hr.toString(),
+              changePercent24Hr: parseFloat(curWithoutId.changePercent24Hr)
+            }
+            return acc
+          } catch {
+            return acc // no caip found, we don't support this asset
+          }
+        }, {} as MarketCapResult)
+    } catch (e) {
+      return {}
+    }
+  }
+
+  findByCaip19 = async ({ caip19 }: MarketDataArgs): Promise<MarketData> => {
+    try {
+      const { tokenId } = fromCAIP19(caip19)
+      const isToken = !!tokenId
+      if (isToken) {
```

Comment:
this confused me at first but then i looked at the coincap api. 

we don't need to pull the tokenId from the caip19 at all, so we can remove this check and throw here.

see my above comment about implementing a coincap caip adapter, so we can turn unique asset ids (caip19s) into coincap ids (strings), then we can just fetch them by coincap id.